src/snowflake/snowpark/stored_procedure_profiler.py

```diff
@@ -122,6 +128,12 @@ def get_output(self) -> str:
         Please call this function right after the stored procedure you want to profile to avoid any error.
 
         """
+        # return empty string when profiler is not enabled to not interrupt user's code
+        if not self._is_enabled:
+            logger.warning(
+                "You are seeing this warning because you try to get profiler output while profiler is disabled. Please use profiler.set_active_profiler() to enable profiler."
+            )
+            return ""
         query_id = self._get_last_query_id()
         if query_id is None:
             raise ValueError("Last executed stored procedure does not exist")
```

Comment:
I don't remember if we discussed this, I prefer log a warning message instead of throwing an exception in this case as well.